backend/lib/endpoints/posts.js

```diff
@@ -127,19 +216,102 @@ async function routes(app) {
 
   app.patch(
     "/:postId",
-    { preValidation: [app.authenticate], schema: updatePostSchema },
+    {
+      preValidation: [app.authenticate],
+      schema: updatePostSchema,
+    },
     async (req, reply) => {
-      // todo: make sure user can only update their own post
+      // TODO: get userId from jwt
+      // userId = ?
+      // user = User.findById(userId);
+      var user = await User.findById("5ea6900c0e0419d4cb123611");
+      if (user === null) return new httpErrors.Unauthorized();
+
       const post = await Post.findById(req.params.postId);
-      if (post === null) {
-        return reply.send(new httpErrors.NotFound());
-      }
-      Object.keys(req.body).forEach((key) => {
-        if (post[key] && post[key] !== req.body[key]) {
-          post[key] = req.body[key];
+      if (!post) return new httpErrors.NotFound();
```

Comment:
I don't think there is an ESLint rule configured for this and we have Prettier so not sure about the rules, but maybe we should be a bit strict about using if / else with returns... here it looks like it can use some more spacing or indentation with `{ }` etc. for readability